apps/emqx/src/emqx_router_sup.erl

```diff
@@ -41,4 +41,9 @@ init([]) ->
         hash,
         {emqx_router, start_link, []}
     ]),
-    {ok, {{one_for_all, 0, 1}, [Helper, RouterPool]}}.
+    SupFlags = #{
+        strategy => one_for_one,
+        intensity => 10,
+        period => 100
+    },
+    {ok, {SupFlags, [Helper, RouterPool]}}.
```

Comment:
Note: we can also set `RouterPool` sup as a significant child, so that its own worker restart intensity is not multiplied with this (`emqx_router_sup`) intensity.. :thinking:

Reply:
Some context about the issue:
the below code in emqx_roter_helper is prone to errors, especially when called on a replicant:
https://github.com/emqx/emqx/blob/master/apps/emqx/src/emqx_router_helper.erl#L155
The reason is that rpc to core node can fail in more or less normal circumstances (e.g. a core node restart). 
Instead of catching the errors inside emqx_router_helper, it seems reasonable to increase intensity to tolerate occasional crashes (probably in some other places as well) . 
Even if the helper fails to delete a record in that case (#L155), another node is expected to do the same, as nodedown message is sent on all the nodes.

Reply:
Since we know some situations can cause the code at line 155 to crash the process, why don't we fix it instead of logging an **incomprehensible Erlang crash error**?
This will cause great panic for our customers, leading to unnecessary trouble.
Perhaps we can catch the error on delete and, if there is an error, directly stop normally to let it restart.
IMO: we should not report avoidable crash logs to customers.

Reply:
If we keep this crash log, it is foreseeable that in the near future, many users will come to inquire about related issues, and we can only tell them that this has no impact. The crash log is useless for us.

Reply:
Thanks @zhongwencool,
Fixed. Actually it crashed with a `badmatch` error, so catching the error is not absolutely required :
```
2023-08-02T06:53:08.917759+00:00 [error] Generic server emqx_router_helper terminating. Reason: {{badmatch,{'EXIT',{aborted,{no_exists,emqx_routing_node}}}},
```
However, I covered and logged both exceptions and error tuples, as there is rather no strict return type guarantee. 
For example, Mnesia dirty operation may exit literally or return `{'EXIT', Reason}` tuple. 
On a replicant side, exit exceptions are not expected as they are normally caught and wrapped by rpc lib, but I'm not sure it won't change in future...

Reply:
After a discussion with @zmstone, we agreed to do router_helper cleanup only on core nodes, as doing it on a replicant is anyway redundant.
Updated the PR.